universum/modules/code_report_collector.py

```diff
@@ -39,30 +32,19 @@ def set_code_report_directory(self, project_root: str) -> None:
         if not os.path.exists(self.report_path):
             os.makedirs(self.report_path)
 
-    def prepare_environment(self, project_configs: List[ProjectConfig]) -> Variations:
-        afterall_steps: List[ProjectConfig] = []
-        for item in project_configs:
-            if not item.get("code_report", False):
+    def prepare_environment(self, project_config_variations: Variations) -> Variations:
+        afterall_steps: Variations = Variations()
+        for item in project_config_variations.configs:
+            if not item.code_report:
                 continue
 
-            self.set_code_report_directory(self.settings.project_root)
-            temp_filename: str = "${CODE_REPORT_FILE}"
-            name: str = utils.calculate_file_absolute_path(self.report_path, item.get("name", "")) + ".json"
+            self.set_code_report_directory(self.settings.project_root)  # why is this in a loop?
```

Comment:
why is this in a loop?

Reply:
my guess it is "legacy reasons"
probably in original version there were no loop

Reply:
anyway, you can move it out of the loop

Reply:
actually, you should move it to after the loop: as I see now, the idea was not to create directory if there are no "code report" steps